### PR TITLE
feat(helm): added a Helm chart holding the deployment configuration f…

### PR DESCRIPTION
**Pull request by pcanilho**
…or [github-token-manager]

### CODE REVIEW

**Comment by pcanilho** [comment]
# Installation

Installed on a local vCluster (EKS) (k8s 1.28):
```console
helm get values -a github-token-manager -n github-token-manager
```

```yaml
COMPUTED VALUES:
crds:
  install: true
deployment:
  containers:
    githubTokenManager:
      image:
        repository: ghcr.io/isometry/github-token-manager
        tag: latest
      resources:
        limits:
          cpu: 500m
          memory: 128Mi
        requests:
          cpu: 5m
          memory: 64Mi
    kubeRbacProxy:
      image:
        repository: gcr.io/kubebuilder/kube-rbac-proxy
        tag: v0.16.0
      resources:
        limits:
          cpu: 500m
          memory: 128Mi
        requests:
          cpu: 5m
          memory: 64Mi
  hostNetwork: false
  nodeSelector: null
  replicas: 1
  tolerations: null
namespace: null
rbac:
  install: true
service:
  ports:
  - name: https
    port: 8443
    protocol: TCP
    targetPort: https
  type: ClusterIP
```

Deployment healthy:
image


CRDs installed:
image

---

# Test

1. Created a Basic & Token CRs on the namespace `lab`:
image

2. Verified that `github-token-manager` is able to populate the `lab` namespace with the necessary secrets:
image

File: deploy/charts/github-token-manager/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+apiVersion: v2
+name: github-token-manager
+description: A Helm chart for Kubernetes
```
Comment (isometry) [inline deploy/charts/github-token-manager/Chart.yaml:4]:
```suggestion
description: A Helm chart for github-token-manager
```

File: deploy/charts/github-token-manager/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,119 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app.kubernetes.io/component: manager
+    app.kubernetes.io/created-by: {{ include "chart.fullname" . }}
+    app.kubernetes.io/instance: manager
+    app.kubernetes.io/managed-by: helm
+    app.kubernetes.io/name: deployment
```
Comment (isometry) [inline deploy/charts/github-token-manager/templates/deployment.yaml:10]: This should probably be `{{ include "chart.name" . }}`

File: deploy/charts/github-token-manager/values.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  install: true
+
+## rbac
+##   create: true | false
+##     true: create the RBAC resources
+##     false: do not create the RBAC resources
+rbac:
+  install: true
+
+## service
```
Comment (isometry) [inline deploy/charts/github-token-manager/values.yaml:19]: Probably worth clarifying here that this is the metrics service, probably even nesting under a `metrics:` top-level key…

File: deploy/charts/github-token-manager/values.yaml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+##     githubTokenManager:
+##       image:
+##         repository: image repository
+##         tag: image tag
+##     kubeRbacProxy:
+##       image:
+##         repository: image repository
+##         tag: image tag
+deployment:
+  hostNetwork: false
```
Comment (isometry) [inline deploy/charts/github-token-manager/values.yaml:48]: If we're going to make this an option, then we need to make *all* the port bindings configurable too…

File: deploy/charts/github-token-manager/values.yaml (hunk 3 above)
Comment (pcanilho) [reply deploy/charts/github-token-manager/values.yaml]: Done!

File: deploy/charts/github-token-manager/values.yaml (hunk 4 above)
Comment (pcanilho) [reply deploy/charts/github-token-manager/values.yaml:59]: Good point. I will remove this instead as adding overrides for all bindings adds unnecessary complexity for now...

**Review by isometry — APPROVED** [review]
LGTM. We can refactor charts into a separate repo "later".

**Pull request merged**